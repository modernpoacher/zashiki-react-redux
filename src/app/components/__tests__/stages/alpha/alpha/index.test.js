import React from 'react'
import Enzyme, { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import Alpha from '@modernpoacher/zashiki-react-redux/app/components/stages/alpha/alpha'

Enzyme.configure({ adapter: new Adapter() })

const MOCK_DEFINITION = {
  meta: {
    uri: '#/'
  },
  elements: {
    title: 'MOCK TITLE',
    field: {
      id: 'MOCK ID'
    }
  }
}

const MOCK_RESOURCE = {
  alpha: 'MOCK ALPHA',
  omega: 'MOCK OMEGA'
}

const MOCK_RESPONSE = {
  '#/': 'MOCK VALUE'
}

const MOCK_ERRORS = []

const MOCK_DEFINITIONS = [
  {
    definition: MOCK_DEFINITION,
    resource: MOCK_RESOURCE,
    response: MOCK_RESPONSE,
    errors: MOCK_ERRORS
  }
]

describe('@modernpoacher/zashiki-react-redux/app/components/stages/alpha/alpha', () => {
  it('renders', () => {
    const component = (
      <Alpha
        definitions={MOCK_DEFINITIONS}
        onChange={jest.fn()}
        onSubmit={jest.fn()}
      />
    )

    expect(toJson(mount(component)))
      .toMatchSnapshot()
  })
})
