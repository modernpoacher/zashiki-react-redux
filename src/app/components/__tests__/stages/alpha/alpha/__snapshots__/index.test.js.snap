// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@modernpoacher/zashiki-react-redux/app/components/stages/alpha/alpha renders 1`] = `
<Alpha
  definitions={
    [
      {
        "definition": {
          "elements": {
            "field": {
              "id": "MOCK ID",
            },
            "title": "MOCK TITLE",
          },
          "meta": {
            "uri": "#/",
          },
        },
        "errors": [],
        "resource": {
          "alpha": "MOCK ALPHA",
          "omega": "MOCK OMEGA",
        },
        "response": {
          "#/": "MOCK VALUE",
        },
      },
    ]
  }
  onChange={[MockFunction]}
  onSubmit={[MockFunction]}
>
  <ol>
    <li
      key="0"
    >
      <form
        action="/MOCK ALPHA/MOCK OMEGA"
        method="POST"
        onSubmit={[Function]}
      >
        <ErrorSummarySprocket
          errorSummary={[]}
          onChange={[Function]}
          title="There is a problem"
        />
        <h2 />
        <Pinion
          onChange={[Function]}
          params={
            {
              "errors": [],
            }
          }
          pinion={
            {
              "elements": {
                "field": {
                  "id": "MOCK ID",
                },
                "title": "MOCK TITLE",
              },
              "meta": {
                "uri": "#/",
              },
            }
          }
        >
          <Field
            elements={
              {
                "field": {
                  "id": "MOCK ID",
                },
                "title": "MOCK TITLE",
              }
            }
            meta={
              {
                "uri": "#/",
              }
            }
            onChange={[Function]}
            params={
              {
                "errors": [],
              }
            }
          >
            <TextCog
              disabled={false}
              name="MOCK ID"
              onChange={[Function]}
              readOnly={false}
              required={false}
              title="MOCK TITLE"
              value=""
            >
              <div
                className="cog text"
              >
                <TextTitle
                  disabled={false}
                  id="MOCK ID"
                  readOnly={false}
                  required={false}
                  title="MOCK TITLE"
                >
                  <label
                    className="title text"
                    htmlFor="MOCK ID"
                  >
                    <TextContent
                      textContent="MOCK TITLE"
                    >
                      <span
                        className="text-content"
                      >
                        MOCK TITLE
                      </span>
                    </TextContent>
                  </label>
                </TextTitle>
                <TextDescription />
                <TextErrorMessage />
                <TextField
                  disabled={false}
                  id="MOCK ID"
                  name="MOCK ID"
                  onChange={[Function]}
                  readOnly={false}
                  required={false}
                  value=""
                >
                  <input
                    className="input text"
                    disabled={false}
                    id="MOCK ID"
                    name="MOCK ID"
                    onChange={[Function]}
                    readOnly={false}
                    required={false}
                    type="text"
                    value=""
                  />
                </TextField>
              </div>
            </TextCog>
          </Field>
        </Pinion>
        <button
          type="submit"
        >
          Continue
        </button>
      </form>
    </li>
  </ol>
</Alpha>
`;
